app/assets/javascripts/discourse/app/models/topic-tracking-state.js

```diff
@@ -65,36 +65,28 @@ const TopicTrackingState = EmberObject.extend({
    * @method establishChannels
    */
   establishChannels() {
-    this.messageBus.subscribe("/new", this._processChannelPayload.bind(this));
-    this.messageBus.subscribe(
-      "/latest",
-      this._processChannelPayload.bind(this)
-    );
-    if (this.currentUser) {
-      this.messageBus.subscribe(
-        "/unread/" + this.currentUser.get("id"),
-        this._processChannelPayload.bind(this)
-      );
-    }
-
-    this.messageBus.subscribe("/delete", (msg) => {
-      this.modifyStateProp(msg, "deleted", true);
-      this.incrementMessageCount();
-    });
-
-    this.messageBus.subscribe("/recover", (msg) => {
-      this.modifyStateProp(msg, "deleted", false);
-      this.incrementMessageCount();
-    });
-
-    this.messageBus.subscribe("/destroy", (msg) => {
-      this.incrementMessageCount();
-      const currentRoute = DiscourseURL.router.currentRoute.parent;
-      if (
-        currentRoute.name === "topic" &&
-        parseInt(currentRoute.params.id, 10) === msg.topic_id
-      ) {
-        DiscourseURL.redirectTo("/");
+    this.messageBus.subscribe("/topic-tracking-state", (data) => {
```

Comment:
I am not familiar with how message bus works; would this create problems with too many messages coming through the same channel, causing delays? Or is the order guaranteed, and messages delivered ASAP? This might not be an issue at all just thinking out loud, the new version is certainly a lot cleaner.